source/fundamentals/linq.txt

```diff
@@ -0,0 +1,541 @@
+         :emphasize-lines: 2
+
+         var query = queryableCollection
+             .OrderBy(r => r.Name)
+             .ThenByDescending(r => r.RestaurantId);
+
+The preceding example returns the query results sorted alphabetically by the
+``Name`` field, with a secondary descending sort on the ``RestaurantId`` field.
+The following is a subset of the returned results:
+
+.. code-block:: json
+
+   // Results Truncated
+
+   ...
+   { "_id" : ObjectId(...), "name" : "Aba Turkish Restaurant", "restaurant_id" : "41548686", "cuisine" : "Turkish", "address" : {...}, "borough" : "Manhattan", "grades" : [...] }
+   { "_id" : ObjectId(...), "name" : "Abace Sushi", "restaurant_id" : "50006214", "cuisine" : "Japanese", "address" : { ... }, "borough" : "Manhattan", "grades" : [...] }
+   { "_id" : ObjectId(...), "name" : "Abacky Potluck", "restaurant_id" : "50011222", "cuisine" : "Asian", "address" : { ... }, "borough" : "Manhattan", "grades" : [...] }
+   { "_id" : ObjectId(...), "name" : "Abaleh", "restaurant_id" : "50009096", "cuisine" : "Mediterranean", "address" : { ... }, "borough" : "Manhattan", "grades" : [...] }
+   ...
+
+$lookup
+~~~~~~~
+
+The ``$lookup`` aggregation stage joins documents from one collection to documents
+from another collection in the same database. The ``$lookup`` stage adds a new 
+array field to each input document. The new array field contains the matching
+documents from the "joined" collection.
+
+.. note:: 
+
+   To perform a lookup, you must make both collections queryable using the
+   ``AsQueryable`` method.
+
+   To learn how to make a collection queryable, see :ref:`csharp-linq-queryable`.
+
+Consider a second collection in the ``sample_restaurants`` database called
+``reviews`` that has restaurant reviews. You can join documents from that collection
+to documents with the same ``name`` value in the ``restaurants`` collection using
+the ``$lookup`` stage.
+
+The following ``Review`` class models the documents in the ``reviews`` collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/linq.cs
+   :language: csharp
+   :dedent:
+   :start-after: start-review-model
+   :end-before: end-review-model
+
+Select the :guilabel:`Query Syntax` or :guilabel:`Method Syntax` tab to see how 
+to generate a ``$lookup`` stage using LINQ:
+
+.. tabs::
+
+   .. tab:: Query Syntax
+      :tabid: query-syntax
+
+      .. code-block:: csharp
+
+         var query = from restaurant in queryableCollection
+             join rv in reviewCollection on restaurant.Name equals rv.RestaurantName into reviews
+             select new { restaurant, reviews };
+
+   .. tab:: Method Syntax
+      :tabid: method-syntax
+
+      .. code-block:: csharp
+
+         var query = queryableCollection
+             .GroupJoin(reviewCollection,
+                 restaurant => restaurant.Name,
+                 review => review.RestaurantName,
+                 (restaurant, reviews) =>
+                     new {restaurant, reviews}
```

Comment:
Is this valid syntax? I was expecting something like `new { Restaurant = restaurant, Revies = reviews }`.

But maybe C# infers the property names?

Reply:
I double checked this and it looks like C# infers the property names, so this executes without issue. I can add the properties in the way you suggested though if that would be a more "correct" (or easier to understand) syntax?

Reply:
Oh interesting about C# inferring the property names. I hadn't used that feature before.

It might be better to use the more normal `Restaurant = restaurant` syntax, specially since the convention in C# is for property names to be capitalized.